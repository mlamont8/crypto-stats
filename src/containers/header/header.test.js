import React from "react";
import { shallow } from "enzyme";
import Header from "./header";


describe("Header", () => {
  it("renders without crashing", () => {
    const snap = shallow(<Header />);
    expect(snap).toMatchSnapshot();
  });
  it("should not show form on initial render", () => {
    const wrapper = shallow(<Header firstLoad />);
    expect(wrapper.find("CoinForm").exists()).toBeFalsy();
  });

  it("should show form after initial render", () => {
    const wrapper = shallow(<Header firstLoad={false} />);
    expect(wrapper.find("CoinForm").exists());
  });
});
