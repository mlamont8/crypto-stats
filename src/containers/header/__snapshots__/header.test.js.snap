// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    firstLoad={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bsStyle": "default",
      "children": Array [
        <NavbarHeader>
          <NavbarBrand>
                    <a
                              href="#brand"
                    >
                              CryptoStats
                    </a>
          </NavbarBrand>
          <NavbarToggle />
</NavbarHeader>,
        null,
      ],
      "collapseOnSelect": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <NavbarBrand>
              <a
                            href="#brand"
              >
                            CryptoStats
              </a>
</NavbarBrand>,
            <NavbarToggle />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <a
                href="#brand"
>
                CryptoStats
</a>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "CryptoStats",
                "href": "#brand",
              },
              "ref": null,
              "rendered": "CryptoStats",
              "type": "a",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      null,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsStyle": "default",
        "children": Array [
          <NavbarHeader>
            <NavbarBrand>
                        <a
                                    href="#brand"
                        >
                                    CryptoStats
                        </a>
            </NavbarBrand>
            <NavbarToggle />
</NavbarHeader>,
          null,
        ],
        "collapseOnSelect": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <NavbarBrand>
                <a
                                href="#brand"
                >
                                CryptoStats
                </a>
</NavbarBrand>,
              <NavbarToggle />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <a
                  href="#brand"
>
                  CryptoStats
</a>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "CryptoStats",
                  "href": "#brand",
                },
                "ref": null,
                "rendered": "CryptoStats",
                "type": "a",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        null,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
